src/storage/validators/QuotaDataValidator.ts

```diff
@@ -0,0 +1,63 @@
+import type { Readable } from 'stream';
+import { PassThrough } from 'stream';
+import type { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { QuotaError } from '../../util/errors/QuotaError';
+import type { Guarded } from '../../util/GuardedStream';
+import { pipeSafely } from '../../util/StreamUtil';
+import type { QuotaStrategy } from '../quota-strategy/QuotaStrategy';
+import type { DataValidator } from './DataValidator';
+
+/**
+ * The QuotaDataValidator validates data streams according to a QuotaStrategy's implementation
+ */
+export class QuotaDataValidator implements DataValidator {
+  private readonly strategy: QuotaStrategy;
+
+  public constructor(strategy: QuotaStrategy) {
+    this.strategy = strategy;
+  }
+
+  public async validateRepresentation(
+    identifier: ResourceIdentifier,
+    data: Guarded<Readable>,
+    metadata: RepresentationMetadata,
+  ): Promise<Guarded<Readable>> {
+    // 1. Get the available size
+    const availableSize = await this.strategy.getAvailableSpace(identifier);
+
+    // 2. Check if the estimated size is bigger then the available size
+    const estimatedSize = await this.strategy.estimateSize(metadata);
+    let checkEstimateSizeHelper = 0;
+    const checkEstimateSize = new PassThrough({
+      // An arrow function cannot have a 'this' parameter.ts(2730)
```

Comment:
And rightly so… shouldn't this be a regular function then? I'm quite surprised it actually works with the `this` pointer in there.